Content.Server/Atmos/Piping/Binary/EntitySystems/GasPressureReliefValveSystem.cs

```diff
@@ -0,0 +1,178 @@
+        and only as much as our max flow rate allows.
+
+        The equations:
+        PV = nRT
+        P1 = P2
+
+        Can be used to calculate the amount of gas we need to transfer.
+        */
+
+        var p1 = inletPipeNode.Air.Pressure;
+        var p2 = outletPipeNode.Air.Pressure;
+
+        if (p1 <= valveEntity.Comp.Threshold || p2 >= p1)
+        {
+            valveEntity.Comp.Enabled = false;
+            _ambientSoundSystem.SetAmbience(valveEntity, false);
+            UpdateAppearance(valveEntity);
+
+            if (valveEntity.Comp.PreviousValveState != valveEntity.Comp.Enabled)
+            {
+                valveEntity.Comp.PreviousValveState = valveEntity.Comp.Enabled;
+                valveEntity.Comp.FlowRate = 0;
+                DirtyFields(valveEntity,
+                    valveEntity.Comp,
+                    MetaData(valveEntity),
+                    nameof(valveEntity.Comp.FlowRate),
+                    nameof(valveEntity.Comp.Enabled));
+            }
+
+            return;
+        }
+
+        // guess we're doing work now
+        valveEntity.Comp.Enabled = true;
+        _ambientSoundSystem.SetAmbience(valveEntity, true);
+        UpdateAppearance(valveEntity);
+
+        var t1 = inletPipeNode.Air.Temperature;
+
+        // First, calculate the amount of gas we need to transfer to bring us below the threshold.
+        var deltaMolesToPressureThreshold =
+            AtmosphereSystem.MolesToPressureThreshold(inletPipeNode.Air, valveEntity.Comp.Threshold);
+
+        // Second, calculate the moles required to equalize the pressure.
+        // Cursed rounding to avoid the floating point error-inator.
+        var deltaMolesToEqualizePressure =
+            float.Round(_atmosphereSystem.FractionToEqualizePressure(inletPipeNode.Air, outletPipeNode.Air) *
+                        inletPipeNode.Air.TotalMoles,
+                digits: 3,
+                MidpointRounding.ToPositiveInfinity);
+
+        // Third, make sure we only transfer the minimum of the two.
+        // We do this so that we don't accidentally transfer so much gas to the point
+        // where the outlet pressure is higher than the inlet.
+        var deltaMolesToTransfer = Math.Min(deltaMolesToPressureThreshold, deltaMolesToEqualizePressure);
+
+        // Fourth, convert to the desired volume to transfer.
+        var desiredVolumeToTransfer = deltaMolesToTransfer * ((Atmospherics.R * t1) / p1);
+
+        // And finally, limit the transfer volume to the max flow rate of the valve.
+        var actualVolumeToTransfer = Math.Min(desiredVolumeToTransfer,
+            valveEntity.Comp.MaxTransferRate * _atmosphereSystem.PumpSpeedup() * args.dt);
+
+        // We remove the gas from the inlet and merge it into the outlet.
+        var removed = inletPipeNode.Air.RemoveVolume(actualVolumeToTransfer);
+        _atmosphereSystem.Merge(outletPipeNode.Air, removed);
+
+        // Calculate the flow rate in L/S for the UI.
```

Comment:
```suggestion
        // Calculate the flow rate in L/s for the UI.
```
fun physics units fact:
capital letters are used for units named after people, lower case for everything else
liter is an exception (at least in the US) as it looks similar to a 1